modules/transports/core/nhttp/src/main/java/org/apache/synapse/transport/nhttp/ClientHandler.java

```diff
@@ -1136,6 +1136,16 @@ private void processResponse(final NHttpClientConnection conn, HttpContext conte
                 && statusCode != HttpStatus.SC_NOT_MODIFIED
                 && statusCode != HttpStatus.SC_RESET_CONTENT) {
             expectEntityBody = true;
+        } else if (NhttpConstants.HTTP_HEAD.equals(requestMethod)) {
+	        // When invoking http HEAD request esb set content length as 0 to response header. Since there is no message
+	        // body content length cannot be calculated inside synapse. Hence additional two headers are added to
+	        // which contains content length of the backend response and the request method. These headers are removed
+	        // before submitting the actual response.
+	        response.addHeader(NhttpConstants.HTTP_REQUEST_METHOD, requestMethod);
+
+	        if (response.getFirstHeader(HTTP.CONTENT_LEN) != null) {
+		        response.addHeader(NhttpConstants.ORIGINAL_CONTENT_LEN, response.getFirstHeader(HTTP.CONTENT_LEN).getValue());
```

Comment:
Thanks Anjana for pointing this out and it fixed with https://github.com/wso2/wso2-synapse/pull/362.